server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -2097,13 +2108,16 @@ void handleBonusPools(Owner owner, Map<String, PoolQuantity> pools,
         public void handlePostEntitlement(PoolManager manager, Consumer consumer,
             Map<String, Entitlement> entitlements) {
             Set<String> stackIds = new HashSet<String>();
+
             for (Entitlement entitlement : entitlements.values()) {
                 if (entitlement.getPool().isStacked()) {
                     stackIds.add(entitlement.getPool().getStackId());
                 }
             }
             List<Pool> subPoolsForStackIds = null;
-            if (!stackIds.isEmpty()) {
+            // Manifest and Share consumers should not contribute to the sharing org's stack,
+            // as Share consumers should not have created a stack derived pool in the first place.
```

Comment:
The comment refers to 'sharing org's stack'. Stacks are not applicable to Manifest/Share consumers. Stacking is a consumer specific feature, not an org's. I just found the comment a little confusing.
Perhaps something along the lines of:

"Stacking does not apply to manifest/shared consumers, so there is no need to check if stack derived pools need updating."

Feel free to modify as you see fit.